src/verify.c

```diff
@@ -308,6 +308,26 @@ static struct list *download_loop(struct list *files, int isfailed)
 		}
 
 		if (hash_needs_work(file, local.hash)) {
+			/* Mix content is local, so don't queue files up for curl downloads */
+			if (file->is_mix) {
+				char *filename;
+				char *url;
+				string_or_die(&url, "%s/%i/files/%s.tar", MIX_STATE_DIR, file->last_change, file->hash);
+				string_or_die(&filename, "%s/download/.%s.tar", state_dir, file->hash);
+				file->staging = filename;
+				ret = link(url, filename);
+				/* Try doing a regular rename if hardlink fails */
+				if (ret) {
+					if (rename(url, filename) != 0) {
+						fprintf(stderr, "Failed to copy local mix file: %s\n", filename);
+						continue;
+					}
+				}
+				untar_full_download(file);
+				free(filename);
+				free(url);
+				continue;
+			}
```

Comment:
I've seen the above hunk in two other places in the diff. Could this be turned into a helper function with a little modification?

Reply:
Yep, it is very similar to the code added to `full_download_loop()`. Not a dealbreaker for now, but a FIXME would be helpful to inform future refactoring.

Reply:
Agree.